packages/react-vapor/src/components/textarea/TextArea.tsx

```diff
@@ -57,43 +58,56 @@ const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITextAreaOwnProps): I
     onUnmount: () => dispatch(removeTextArea(ownProps.id)),
 });
 
-export class TextArea extends React.Component<ITextAreaProps, {}> {
-    static defaultProps: Partial<ITextAreaOwnProps> = {
-        additionalAttributes: {},
-        className: '',
-    };
+export const TextArea = (props: ITextAreaProps) => {
+    const [isValid, setIsValid] = React.useState(true);
 
-    componentWillMount() {
-        if (this.props.onMount) {
-            this.props.onMount();
-        }
-    }
+    React.useEffect(() => {
+        setIsValid(!!props.validate && props.validate(props.value));

```

Comment:
To avoid lags, we could debounce the validation. Otherwise, it will validate and rerender each time the user enters a character. With a deboune, it will wait until a given amount of time (say 300 ms) before triggering the validation.